### DISCUSSION TITLE
References to computationally-generated sub-tables/sub-figures

### DISCUSSION THREAD

**Discussion by connorp** [quarto-dev/quarto-cli#1494] [Q&A] [answered]
Per the docs, we can create sub-tables and sub-figures with captions and _cross-references_ if we create them [manually](https://quarto.org/docs/authoring/tables.html#subtables). We can also create sub-tables and sub-figures programmatically, including specifying [sub-captions](https://quarto.org/docs/authoring/tables.html#computations). However, it's not clear to me whether I can _reference_ those sub-figures/tables. I tried something like this:

````
```{r}
#| label: many-tables
#| tbl-cap: So many tables!
#| tbl-subcap: 
#|   - Table One {#tbl-one}
#|   - Table Two {#tbl-two}
#|   - Table Three {#tbl-three}

... # some code

```
````

But that did not succeed. Is this possible currently, and if not, can we come up with a way to do that?

**Comment by jjallaire** [comment, marked as answer]
Yikes, I don't think we actually show that in the docs for tables! Analogous to how we handle figures (which is documented, https://quarto.org/docs/authoring/cross-references.html#computations) we auto-generate the sub-table labels off of a master label. So for example:

````markdown
```{r}
#| label: tbl-so-many
#| tbl-cap: So many tables!
#| tbl-subcap: 
#|   - Table One 
#|   - Table Two 
#|   - Table Three

... # some code

```

See @tbl-so-many-1 and @tbl-so-many-2

````

Note that you need to use the `tbl` prefix in your label to create a cross-referencable table.

**Reply by connorp** [reply]
ah, perfect. thanks!
